dags/python_scripts/atd_vzd_cr3_scan_pdf_records.py

```diff
@@ -0,0 +1,300 @@
+    """
+    Downloads a CR3 PDF file into disk
+    :param int crash_id: The crash id of the record to be downloaded
+    :return bool: True if it succeeds, false otherwise.
+    """
+    try:
+        if is_crash_id(crash_id):
+            S3_CLIENT.download_file('atd-vision-zero-editor', f"production/cris-cr3-files/{crash_id}.pdf", f"{crash_id}.pdf")
+            return file_exists(crash_id)
+        else:
+            return False
+    except:
+        return False
+
+
+def delete_file(crash_id: int) -> bool:
+    """
+    Deletes a CR3 PDF file from disk
+    :param int crash_id: The crash id of the file to be deleted
+    :return bool:
+    """
+    if is_crash_id(crash_id) and file_exists(crash_id):
+        os.remove(f"./{crash_id}.pdf")
+        return file_exists(crash_id) is False
+    else:
+        return False
+
+
+def get_mime_attributes(crash_id: int) -> dict:
+    """
+    Runs a shell command and returns a dictionary with mime attributes
+    :return dict:
+    """
+    # Make sure the id is an integer and that the file exists...
+    if not is_crash_id(crash_id) or not file_exists(crash_id):
+        return {}
+
+    # Execute command to get mime attributes as accurately as possible:
+    try:
+        command = PDF_MIME_COMMAND.substitute(PDF_FILE=f"./{crash_id}.pdf")
+        mime_attr_str = os.popen(command).read()
+        mime_attr = mime_attr_str.replace(" charset=", "").strip().split(";", 2)
+
+        return {
+            "mime_type": mime_attr[0],
+            "encoding": mime_attr[1]
+        }
+    except IndexError:
+        return {}
+
+
+def get_file_size(crash_id: int) -> int:
+    """
+    Gets the file size for crash_id
+    :param int crash_id: The crash id of the file to be tested
+    :return int:
+    """
+    if not is_crash_id(crash_id) or not file_exists(crash_id):
+        return 0
+
+    try:
+        return os.path.getsize(f"./{crash_id}.pdf")
+    except FileNotFoundError:
+        return 0
+
+
+def get_timestamp() -> str:
+    """
+    Returns a string containing the date and time the file was tested
+    :return str:
+    """
+    return datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+
+
+def get_file_metadata(crash_id: int) -> dict:
+    """
+    Returns a dictionary containing the metadata for a file. It assumes the file already exists in disk.
+    :param int crash_id:
+    :return dict:
+    """
+
+    if not is_crash_id(crash_id) or not file_exists(crash_id):
+        return {}
+
+    timestamp = get_timestamp()
+    file_size = get_file_size(crash_id)
+    mime_attr = get_mime_attributes(crash_id)
+
+    return {
+        "last_update": timestamp,
+        "file_size": file_size,
+        "mime_type": mime_attr.get("mime_type", None),
+        "encoding": mime_attr.get("encoding", None)
+    }
+
+
+def process_record(crash_id: int) -> bool:
+    """
+    Controls the process for a single file
+    """
+    if not is_crash_id(crash_id):
+        print(f"Invalid crash_id: {crash_id}")
```

Comment:
any reason not to use `logging` instead of `print` statements?

Reply:
I know that print will definitely show in the Airflow logs, logging I think you have to pipe the output or change the configuration in Airflow.

Reply:
yes, was debugging airflow logging over the weekend 😑 .  i haven't tested with BashOperator, but [this](https://github.com/cityofaustin/atd-knack-services/blob/50362ae4f3a339072de51faa68fae85326b2ea9a/services/records_to_s3.py#L86) fixed my logging with DockerOperator